Introduce FileSlice

wiktorn opened the pull request:
New class providing "windowed" view over the file objects. It gives the
possibility to have a file-alike object that returns only some part of the file.
Useful for multipart upload.

wiktorn committed:
Introduce FileSlice

New class providing "windowed" view over the file objects. It gives the
possibility to have a file-alike object that returns only some part of the file.
Useful for multipart upload.

wiktorn committed:
Introduce FileSlice

New class providing "windowed" view over the file objects. It gives the
possibility to have a file-alike object that returns only some part of the file.
Useful for multipart upload.

wiktorn committed:
Introduce FileSlice

New class providing "windowed" view over the file objects. It gives the
possibility to have a file-alike object that returns only some part of the file.
Useful for multipart upload.

msftclas commented:
Hi **@wiktorn**, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

wiktorn committed:
Introduce fragment_upload

Introduce new function that is handling all the hassles of multipart-upload.
Creates upload session, uploads the file in recommended 10MB parts,
refreshes authentication token if needed.

Sample usage:
client.item(drive="me", path="foo/bar.txt").fragment_upload("/path/to/file")

wiktorn committed:
Introduce fragment_upload

Introduce new function that is handling all the hassles of multipart-upload.
Creates upload session, uploads the file in recommended 10MB parts,
refreshes authentication token if needed.

Sample usage:
client.item(drive="me", path="foo/bar.txt").fragment_upload("/path/to/file")

wiktorn committed:
Introduce fragment_upload

Introduce new function that is handling all the hassles of multipart-upload.
Creates upload session, uploads the file in recommended 10MB parts,
refreshes authentication token if needed.

Sample usage:
client.item(drive="me", path="foo/bar.txt").fragment_upload("/path/to/file")

wiktorn committed:
Introduce fragment_upload

Introduce new function that is handling all the hassles of multipart-upload.
Creates upload session, uploads the file in recommended 10MB parts,
refreshes authentication token if needed.

Sample usage:
client.item(drive="me", path="foo/bar.txt").fragment_upload("/path/to/file")

wiktorn committed:
Allow sending file-like objects in send

wiktorn committed:
Allow sending file-like objects in send

wiktorn committed:
Merge branch 'fragment-upload' of https://github.com/wiktorn/onedrive-sdk-python into fragment-upload

cdmayer commented on src/onedrivesdk/extensions/fragment_upload.py:
```diff
@@ -0,0 +1,176 @@
+'''
+The MIT License (MIT)
+
+Copyright (c) 2015 Wiktor Niesiobędzki
```
Actually, in the CLA it says you can't keep the copyright. Please copy/paste the license out of one of the other files as-is.

cdmayer commented:
Once you fix the licenses this should be good to go!

wiktorn committed:
Merge branch 'fragment-upload' of https://github.com/wiktorn/onedrive-sdk-python into fragment-upload

wiktorn commented:
Please wait for #11, before commiting this, as for now - this code requires changes in #11, as it calls create_session() without arguments.

wiktorn committed:
Merge branch 'fragment-upload' of https://github.com/wiktorn/onedrive-sdk-python into fragment-upload

="foo/bar.txt").fragment_upload("/path/to/file")

Introduce fragment_upload

Introduce new function that is handling all the hassles of multipart-upload.
Creates upload session, uploads the file in recommended 10MB parts,
refreshes authentication token if needed.

Sample usage:
client.item(drive="me", path="foo/bar.txt").fragment_upload("/path/to/file")

wiktorn committed:
Add optional argument for name conflict resultion in fragment upload

wiktorn commented:
Ok, so now it's independent of #11. Though I'm not sure, if my intensive rebasing will not anyway pollute the commit history

cdmayer commented:
I appreciate it nonetheless. This one is approved.

dmrickert commented:
+1 for getting this into master
